domain/src/main/java/life/qbic/apps/datamanager/services/UserRegistrationService.java

```diff
@@ -36,42 +55,217 @@ public UserRegistrationService(NotificationService notificationService, EventSto
+      }
+
+      @Override
+      public void handleEvent(UserRegistered event) {
+        eventStore.append(event);
+        sendNotification(event);
+      }
+
+      private void sendNotification(UserRegistered event) {
+        var notificationId = notificationService.newNotificationId();
+        var notification =
+            Notification.create(
+                event.getClass().getSimpleName(),
+                event.occurredOn(),
+                notificationId,
+                event);
+        notificationService.send(notification);
+      }
+    });
+
+    var userEmail = Email.from(email);
+    var userFullName = FullName.from(fullName);
+    var userPassword = EncryptedPassword.from(rawPassword);
+
+    if (userRepository.findByEmail(userEmail).isPresent()) {
+      return RegistrationResponse.failureResponse(new UserExistsException());
+    }
+
     // Trigger the user creation in the domain service
-    userDomainService.get().createUser(fullName, email, rawPassword);
+    userDomainService.get().createUser(userFullName, userEmail, userPassword);
+
     // Overwrite the password
     Arrays.fill(rawPassword, '-');
+    return RegistrationResponse.successResponse();
+  }
+
+  private RegistrationResponse validateInput(String fullName, String email, char[] rawPassword) {
+    List<Exception> failures = new ArrayList<>();
+
+    try {
+      Email.from(email);
+    } catch (EmailValidationException e) {
+      failures.add(e);
+    }
+    try {
+      FullName.from(fullName);
+    } catch (Exception e) {
+      failures.add(e);
+    }
+    try {
+      EncryptedPassword.from(rawPassword);
+    } catch (PasswordValidationException e) {
+      failures.add(e);
+    }
+
+    if (failures.isEmpty()) {
+      return RegistrationResponse.successResponse();
+    }
+
+    if (failures.size() > 1) {
+      return RegistrationResponse.failureResponse(failures.get(0),
+          failures.subList(1, failures.size())
+              .toArray(failures.toArray(new Exception[failures.size() - 1])));
+    } else {
+      return RegistrationResponse.failureResponse(failures.get(0));
+    }
+  }
+
+  /**
+   * Activates a user with the userId provided. If no user is matched then this method does
+   * nothing.
+   *
+   * @param userId the id of the user to be activated
+   * @throws UserNotFoundException when no user with the provided user id can be found.
+   * @since 1.0.0
+   */
+  public void confirmUserEmail(String userId) throws UserNotFoundException {
+    DomainEventPublisher.instance().subscribe(new DomainEventSubscriber<UserActivated>() {
+      @Override
+      public Class<UserActivated> subscribedToEventType() {
+        return UserActivated.class;
+      }
+
+      @Override
+      public void handleEvent(UserActivated event) {
+        eventStore.append(event);
+        sendNotification(event);
+      }
+
+      private void sendNotification(UserActivated event) {
+        var notificationId = notificationService.newNotificationId();
```

Comment:
This will cause a massive merge conflict when the feature branch for email confirmation will be merged.